src/slave/slave.cpp

```diff
@@ -3656,29 +3656,38 @@ void Slave::launchExecutor(
       framework->info.checkpoint());
 
 
-  if(environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled"){
-    // Slave has CNI enabled but doesn't define any networks, the criteo subnet must be injected.
-    ContainerInfo* containerInfo = containerConfig.mutable_container_info();
-    containerInfo->set_type(ContainerInfo_Type::ContainerInfo_Type_MESOS);
-
-    if (containerInfo->network_infos_size() == 0) {
-      NetworkInfo* criteoNet = containerInfo->add_network_infos();
-      criteoNet->set_name("criteo");
-
-      if(containerConfig.mutable_executor_info()->has_discovery()){
-        // We need to take the defined ports and map them in the public subnet
-        Ports ports = *containerConfig.mutable_executor_info()
-                  ->mutable_discovery()
-                  ->mutable_ports();
-
-        for (const auto port : *ports.mutable_ports()) {
-          auto p = criteoNet->add_port_mappings();
-          p->set_host_port(port.number());
-          p->set_container_port(port.number());
-          p->set_protocol(port.protocol());
-        }
+  const vector<string> cniNetworks;
+  if (environment.find("CRITEO_NETWORKS") != environment.end()) {
+      cniNetworks = strings::split(environment["CRITEO_NETWORKS"], ",");
+  } else if (environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled") {
+      cniNetworks << "criteo"
+  }
+  if (!cniNetworks.empty()) {
+      // Slave has CNI enabled but doesn't define any networks, the criteo subnet must be injected.
```

Comment:
let's update the comment to be more concise following this rework: "default subnets need to be injected"
OPT: rename slave by agent

Reply:
updated